src/features/pastelStatistics/components/chart/EChartsLineChart.tsx

```diff
@@ -0,0 +1,307 @@
+import * as echarts from 'echarts'
+import ReactECharts from 'echarts-for-react'
+import { saveAs } from 'file-saver'
+import * as htmlToImage from 'html-to-image'
+import React, { useEffect, useRef, useState } from 'react'
+import { CSVLink } from 'react-csv'
+import { Data } from 'react-csv/components/CommonPropTypes'
+import { TPeriod, makeDownloadFileName } from '../../utils/PastelStatisticsLib'
+
+import styles from './LineChart.module.css'
+
+type TThemeColor = {
+  name: string
+  backgroundColor: string
+  lineColor?: string
+  splitLineColor: string
+  color: string
+  stack?: string
+  smooth?: boolean
+}
+
+type LineChartProps = {
+  dataX?: string[]
+  dataY?: number[]
+  title?: string
+  periodIndex: number
+  handleBgColorChange: (color: string) => void
+  handlePeriodFilterChange?: (period: TPeriod) => void
+}
+
+export const EChartsLineChart = (props: LineChartProps): JSX.Element => {
+  const [csvData, setCsvData] = useState<string | Data>('')
+  const csvHeaders = [
+    { label: 'Value', key: 'value' },
+    { label: 'Created time', key: 'time' },
+  ]
+  const downloadRef = useRef(null)
+  const [selectedPeriodButton, setSelectedPeriodButton] = useState(0)
+  const [selectedThemeButton, setSelectedThemeButton] = useState(0)
+  const [currentTheme, setCurrentTheme] = useState<TThemeColor | null>()
+  const [eChartRef, setEChartRef] = useState<ReactECharts | null>()
+  const [eChartInstance, setEChartInstance] = useState<echarts.ECharts>()
+  const {
+    dataX,
+    dataY,
+    title,
+    periodIndex,
+    handlePeriodFilterChange,
+    handleBgColorChange,
+  } = props
+  const [minY, setMinY] = useState(0)
+  const [maxY, setMaxY] = useState(0)
+  const periods: TPeriod[][] = [
+    ['2h', '2d', '4d', 'all'],
+    ['30d', '60d', '180d', '1y', 'all'],
+  ]
+  const themes = [
```

Comment:
We are using the same code in the file `EChartsMultiLineChart.tsx`. Can we move them to the common folder?